notes.md

```diff
@@ -0,0 +1,264 @@
+```python
+Processing post_reads_per_date with args 12, 2022-01-24
+Cache before: defaultdict(<class 'dict'>, {})
+Cache after: defaultdict(<class 'dict'>, {datetime.date(2022, 1, 24): {12: 1}})
+```
+
+
+## Listening to notifications from triggers
+Here the user wishes to ensure that whenever an `Author` object is created, a corresponding
+`Post` object is created which references the newly created `Author`.
+
+### Declaring a listener
+As before, the user declares the listener in their
+apps `listeners.py` file.
+We can make use of the high-level `pgpubsub.listen.post_insert_listen` decorator here.
+```python
+@post_insert_listen(channel_name='author_insert', model=Author)
+def create_first_post_for_author(trigger_payload: TriggerPayload):
+    new_author = trigger_payload.new
+    print(f'Creating first post for {new_author.name}')
+    Post.objects.create(
+        author_id=new_author.pk,
+        content='Welcome! This is your first post',
+        date=datetime.date.today(),
+    )
+```
+This does two things: registers a listener  and installs
+a trigger.
+
+#### Registering the listener
+This is very similar to the above example. When the project is loaded,
+the `post_insert_listen` decorator is called. This
+executes this piece of logic:
+```python
+TriggerPayloadChannel.register(callback, channel_name)
+```
+The `TriggerPayloadChannel` is a channel object as described in the previous
+example. The only difference between this subclass and the `CustomPayloadChannel`
+channel subclass used earlier is the deserialization logic:
+- As explained,
+  `CustomPayloadChannel` is used to deserialize to match the signature of a custom
+  callback,
+- `TriggerPayloadChannel` is used to deserialize the payload which comes
+  from an notification which is the result of a trigger. These notifications
+  are of course invoked at the SQL level, so the user does not directly
+  control what they look like. This is why we insist that any "trigger listener"
+  (like `create_first_post_for_author`) above has the a signature of taking
+  one argument of type `TriggerPayload`. It is thus the responsibility
+  of `TriggerPayloadChannel` to deserialize the trigger notification payload
+  to a `TriggerPayload` instance (which then has access to `old` and `new` objects).
+
+#### Installing the trigger
+
+The `post_insert_listen` decorator instantiates `Notify` object
+as follows
+```python
+def post_insert_listen(model, channel_name=None):
+    return trigger_listen(
+        model,
+        trigger=Notify(
+            name=channel_name,
+            when=pgtrigger.After,
+            operation=pgtrigger.Insert,
+        ),
+        channel_name=channel_name,
+    )
+```
+where `Notify` is a subclass of `pgtrigger.Trigger.
+This is then installed in `trigger_listen` function:
+```python
+trigger.install(model)
+```
+I'm unsure if this is the best practice for installing
+triggers programatically.
```

Comment:
This is a totally valid way of doing it! I believe you may have to manually call pgtrigger.register or something like that, but this should work fine